pull_request_template.md

```diff
@@ -0,0 +1,10 @@
+## Describe your changes
+
+## Relevant Jira
+
+## Checklist before requesting a review
+- [ ] I have marked as draft or added `depends on` label if there's a dependency PR
```

Comment:
What is a `depends on` label? Is that something that Github knows about and will block on? I did a quick google and it seems it's something that people requested, but probably not something that Github supports? So I wonder how it would really help - other than perhaps the person merging the PR may notice.

So I think if it doesn't really do anything, opening a thread somewhere appropriate is probably more reliable - open threads do block merging of PRs.

Reply:
I thought we could make any custom label we wanted. Maybe `donotmerge` would be a better label.  People have created workflows to prevent merging [conditionally based on labels. ](https://stackoverflow.com/questions/71502652/prevent-merging-of-branch-conditionally-based-on-label)

But really I think it's just more of a visual reminder for the maintainers of this repo.  `draft` would accomplish the same thing and actually blocks, so that's probably just as good.

Reply:
You can add reviewers to a draft PR though. I do this often

Reply:
Totally. But for a newcomer / external contributor, it might not be obvious whom to add.

Reply:
release-service-maintainers. The same as is auto added if a PR is opened. If they want review on their draft, they can add the whole team or the specific person they want

Reply:
I know this of course. But an external contributor may not.

Reply:
What I am advocating for is to add it to this document. "If you want reviews on your draft PR, you can add reviewers or add the handle if you are unsure who to tag"